File: server/structs/printer.go
package structs

type PrintSettings struct {
	TargetNozzleTemp int `json:"target_nozzle_temp,omitempty"`
	TargetHotbedTemp int `json:"target_hotbed_temp,omitempty"`
	FanSpeedPct      int `json:"fan_speed_pct,omitempty"`
	PrintSpeedMode   int `json:"print_speed_mode,omitempty"`
	Zcompensation    int `json:"z_comp,omitempty"`
}

type PrintState struct {
	// Print state
	Taskid        string `json:"taskid,omitempty"`
	Filename      string `json:"filename,omitempty"`
	PrintTime     int    `json:"print_time,omitempty"`
	SuppliesUsage int    `json:"supplies_usage,omitempty"`
	TotalLayers   int    `json:"total_layers,omitempty"`
	CurrLayer     int    `json:"curr_layer,omitempty"`
	RemainTime    int    `json:"remain_time,omitempty"`
}

File: kobra-client/src/interfaces/printer.ts
export interface PrinterState {
  state: string;
  currentNozzleTemp: number;
  currentBedTemp: number;
  targetNozzleTemp: number;
  targetBedTemp: number;
  fanSpeed: number;
  printSpeed: number;
  zComp: string;
}

export interface ITempColor {
  // Nozzle and bed temp
  nozzle: string;
  bed: string;
  // Fan speed
  fan: string;
  status: string;
  zComp: string;
}

export interface IPrintJob {
  taskid?: string;
  filename?: string;
  print_time?: number;
  progress?: number;
  supplies_usage?: number;
  total_layers?: number;
  curr_layer?: number;
  remain_time?: number;
}

export interface IFile {
  name: string;
  size: number;
  modified_at: string;
  path: string;
}
